pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1440,6 +1440,27 @@ self: super: {
     HsYAML = self.HsYAML_0_2_1_0;
   };
 
+  # this will probably need to get updated with every ghcide update,
+  # we need an override because ghcide is tracking haskell-lsp closely.
+  ghcide = dontCheck (super.ghcide.override rec {
+    haskell-lsp-types = super.callHackageDirect {
+      pkg = "haskell-lsp-types";
+      ver = "0.19.0.0";
+      sha256 = "1z0c9c2zjb4ad3ffzng9njyn9ci874xd8mmqwnvnm3hyncf1430g";
+    } {};
+    haskell-lsp = super.callHackageDirect {
+      pkg = "haskell-lsp";
+      ver = "0.19.0.0";
+      sha256 = "1v0r57g2dhradnjnvp40jmv5swawg9k3d735kj50dca1gbx66y0c";
+    } {
+      inherit haskell-lsp-types;
+    };
```

Comment:
I don't think `callHackageDirect` will work in nixpkgs because it uses IFD.

Alternatively, you can use `cabal2nix` to create a derivation for `haskell-lsp-types_0_19_0_0` and `haskell-lsp_0_19_0_0` by hand and throw it in `pkgs/development/haskell-modules/hackage-packages.nix`.

The `regex-tdfa` and `haddock-library` overrides look fine.

Reply:
Oh, I didn‘t realize the IFD problem. Okay.
But isn‘t `hackage-packages.nix` autogenerated? Won‘t it get overriden? Is that really the right place to put it?

Reply:
Mind you. Probably it won‘t get overwritten on stable anymore. But I also need a solution that works on the haskell-updates branch.

Reply:
Thanks for the feedback. I guess I figured out that I also need to add it to extra-packages in the hackage2nix.yaml so that it does not get overriden.

Reply:
Unfortunately there isn't a single solution that works for both the release branches and `haskell-updates`.

On the release branches, `hackage-packages.nix` is not autogenerated anymore.  It should be okay to edit it directly.

On `haskell-updates`, the way to get new packages generated is to add the package to this list:

https://github.com/NixOS/nixpkgs/blob/0a115f808755ad6eec5eb7742c71d02e4d0f9d9f/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L2361-L2370

I have somewhat of an explanation of this process here:

https://discourse.nixos.org/t/haskellpackages-stm-containers-fails-to-build/5416/4

Reply:
Okay. Well, now I tried to do a change that fits for both branches. i.e. change nix file and the yaml file both. I can amend my commits so that I do the one for the release and the other for haskell-updates.
But not change the .nix file for haskell-updates is annoying because then I can‘t even test if it‘s working right now. Or is it desired for me to run hackage2nix after changing the yaml?

Reply:
> But not change the .nix file for haskell-updates is annoying because then I can‘t even test if it‘s working right now. Or is it desired for me to run hackage2nix after changing the yaml?

I don't have a good suggestion for you.

However, there are people that are working on a way to make running `hackage2nix` easier:

https://github.com/NixOS/nixpkgs/pull/86699

-------------------------------------

One easy(?) thing you could do is send a PR adding the packages you need to `extra-packages`, wait for it to be merged in and hackage-packages.nix regenerated, and then send a second PR using those newly regenerated packages.